src/figuresClasses.ts

```diff
@@ -1,19 +1,81 @@
-export interface Figure {
+type Shape = 'triangle' | 'circle' | 'rectangle';
+type Color = 'red' | 'green' | 'blue';
 
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  public shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0) {
+      throw new Error('Length of sides must be greater than 0');
+    }
+
+    if (a + b <= c || b + c <= a || c + a <= b) {
+      throw new Error(
+        `Sides ${this.a}, ${this.b} and ${this.c} cannot form a triangle`,
+      );
```

Comment:
good job - clear and concise 🔥 
To make it perfect - move this logic to a separate method and just call this method inside the constructor